airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/buffered_stream_consumer/BufferedStreamConsumer.java

```diff
@@ -154,18 +156,20 @@ protected void acceptTracked(AirbyteMessage message) throws Exception {
     }
 
     // ignore other message types.
-    final AirbyteStreamNameNamespacePair pair = AirbyteStreamNameNamespacePair.fromRecordMessage(message.getRecord());
+    AirbyteRecordMessage record = message.getRecord();
+    final AirbyteStreamNameNamespacePair pair = AirbyteStreamNameNamespacePair.fromRecordMessage(record);
     if (!pairs.contains(pair)) {
       throw new IllegalArgumentException(
           String.format("Message contained record from a stream that was not in the catalog. \ncatalog: %s , \nmessage: %s",
               Jsons.serialize(catalog), Jsons.serialize(message)));
     }
-    var data = Jsons.serialize(message.getRecord().getData());
+    var data = Jsons.serialize(record.getData());
```

Comment:
Why do you need to serialize instead of having RecordData contain a JsonNode? 

That will prevent having a double serialization.

Reply:
this serialisation is so we can check the Json string's length. I thought of comparing byte array length but that seemed overly complicated. open to suggestion to a better way of doing this.